File: Backend/app/models/book.py
from dataclasses import dataclass
from pydantic import BaseModel

# The Book class represents a book object with attributes 'id', 'title', and 'state'.
@dataclass
class Book:
    # 'id' is an integer representing the unique identifier of the book.
    id: int
    
    # 'title' is a string representing the title of the book.
    title: str
    
    # 'state' is a string representing the current state or status of the book.
    state: str

File: src/app/models/book.ts
// Define an interface named 'Book' to represent book objects.
export interface Book {
  // 'id' is a unique identifier for the book, typically a number.
  id: number;

  // 'title' is a string that represents the title of the book.
  title: string;

  // 'state' is a string that represents the state or status of the book.
  // This could be used to indicate whether the book is in progress, completed, or some other state.
  state: string;
}
